sia/src/rhp/quic.rs

```diff
@@ -96,58 +100,21 @@ impl Transport for Stream {
 
 /// A Dialer manages QUIC connections to Sia hosts.
 /// Connections will be cached for reuse whenever possible.
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub struct Dialer {
-    /*
-        note: quinn's documentation (https://docs.rs/quinn/latest/quinn/struct.Endpoint.html#method.client) suggests
-        non-ideal fallback behavior when dual-stack is not supported. This effectively treats every platform as
-        single-stack instead since IPv4 is the preferred fallback.
-    */
-    endpoint_v4: Option<Endpoint>,
-    endpoint_v6: Option<Endpoint>,
-
-    hosts: Mutex<HashMap<PublicKey, Vec<NetAddress>>>,
-    open_conns: Arc<Mutex<HashMap<PublicKey, Connection>>>,
-    cached_prices: Mutex<HashMap<PublicKey, HostPrices>>,
+    inner: Arc<DialerInner>,
```

Comment:
Tbh I don't love the "inner" pattern but I guess it's alright if we exclusively want to access a type through an `Arc`. Wrapping `Self` in an `Arc` on every method would probably be worse.